new-frontiers-14/frontier-station-14
# Mail Metrics: access from anywhere, consistent error tracking.

## About the PR



Adds a sector service for mail logistics and tracking.  The PDA app can be checked from anywhere in-game and receive accurate statistics.

Additionally, errors are only tracked for the first issue that happens to a package (when it no longer becomes profitable).  We care why a package's handling led to lost revenue, not that a package was disrespected after no longer becoming profitable.

## Why / Balance


Mostly in the about sections, but UI inconsistency.

## How to test


1. Spawn as mail carrier or SR (or install the app through cartridge)
2. Purchase a ship.
3. Move onto the ship, check your PDA mail metrics app.  You should not have a blank screen.
4. Move to the mail teleporter.
5. View variables on the teleporter, go to the server-side MailTeleporterComponent.
6. Edit priorityDuration to 10 seconds.
7. Run `mailnow` until you get priority mail.
8. Wait for the priority mail to lapse.
9. Spawn an emag, emag the mail.
10. Run `mailnow` until you get priority mail.
11. Wait for the priority mail to lapse.
12. Throw the package against the wall until it breaks.
13. Check your app.  The only failure should be for the expired priority mail, not for breaking or emagging the packages.

## Media


TBD

## Breaking changes


**Changelog**



No changelog, not yet in-game.
